### ISSUE TITLE
Error in 2.3.0-preview-0037-gad595e44ed: Could not load file or assembly System.Collections

### ISSUE THREAD

**Issue by Bosch-Eli-Black** [mono/t4#123]
We just upgraded from 2.1.0 to 2.3.0-preview-0037-gad595e44ed and are now seeing the following error:
```
C:\ex.csproj(66,3): error : Processing 'Example\GenerateCrud.t4' failed.
C:\ex.csproj(66,3): error : ERROR: Error running transform: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
C:\ex.csproj(66,3): error :  ---> System.IO.FileLoadException: Could not load file or assembly 'System.Collections, Version=4.0.10.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The parameter is incorrect. (0x80070057 (E_INVALIDARG))
C:\ex.csproj(66,3): error : File name: 'System.Collections, Version=4.0.10.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
C:\ex.csproj(66,3): error :  ---> System.ArgumentException: Path "System.Collections.dll" is not an absolute path. (Parameter 'assemblyPath')
C:\ex.csproj(66,3): error :    at System.Runtime.Loader.AssemblyLoadContext.LoadFromAssemblyPath(String assemblyPath)
C:\ex.csproj(66,3): error :    at System.Runtime.Loader.AssemblyLoadContext.ResolveUsingLoad(AssemblyName assemblyName)
C:\ex.csproj(66,3): error :    at System.Runtime.Loader.AssemblyLoadContext.Resolve(IntPtr gchManagedAssemblyLoadContext, AssemblyName assemblyName)
C:\ex.csproj(66,3): error :    at SharpYaml.Serialization.YamlStream..ctor()
C:\ex.csproj(66,3): error :    at Microsoft.OpenApi.Readers.OpenApiTextReaderReader.LoadYamlDocument(TextReader input) in D:\a\1\s\src\Microsoft.OpenApi.Readers\OpenApiTextReaderReader.cs:line 87
C:\ex.csproj(66,3): error :    at Microsoft.OpenApi.Readers.OpenApiTextReaderReader.Read(TextReader input, OpenApiDiagnostic& diagnostic) in D:\a\1\s\src\Microsoft.OpenApi.Readers\OpenApiTextReaderReader.cs:line 43
C:\ex.csproj(66,3): error :    at Microsoft.OpenApi.Readers.OpenApiStreamReader.Read(Stream input, OpenApiDiagnostic& diagnostic) in D:\a\1\s\src\Microsoft.OpenApi.Readers\OpenApiStreamReader.cs:line 37
C:\ex.csproj(66,3): error :    at Microsoft.VisualStudio.TextTemplatingc91ce8a.GeneratedTextTransformation.TransformText() in C:\sources_git\strata\StrataMaker\StrataMaker\CodeGen\GenerateApiControllers.t4:line 16
C:\ex.csproj(66,3): error :    --- End of inner exception stack trace ---
C:\ex.csproj(66,3): error :    at System.RuntimeMethodHandle.InvokeMethod(Object target, Span`1& arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
C:\ex.csproj(66,3): error :    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
C:\ex.csproj(66,3): error :    at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
C:\ex.csproj(66,3): error :    at Mono.TextTemplating.CompiledTemplate.Process() in /_/Mono.TextTemplating/Mono.TextTemplating/CompiledTemplate.cs:line 163
C:\ex.csproj(66,3): error MSB3073: The command "dotnet tool run t4 Example\GenerateCrud.t4 --out Example\GenerateCrud.t4.cs --verbose -I C:\ex\CodeGen -p:openApiPath="C:\ex\example-1.0.openapi.json" -p:apiRoot="Example" -p:apiNamespace="Example" -p:libraryInterface="Example" -p:libraryInterfaceNamespace="Example" -r "%userprofile%\.nuget\packages\microsoft.openapi\1.2.3\lib\netstandard2.0\Microsoft.OpenApi.dll" -r "%userprofile%\.nuget\packages\microsoft.openapi.readers\1.2.3\lib\netstandard2.0\Microsoft.OpenApi.Readers.dll" -r "%userprofile%\.nuget\packages\sharpyaml\1.6.5\lib\netstandard2.0\SharpYaml.dll"" exited with code 1.
```

Not sure if this is a regression or a bug on our end.

Here's the `.t4` file that causes the error:
```t4
<#@ template debug="true" hostspecific="true" language="C#" #>
<#@ include file="OpenApiHeader.t4" #>
<#@ parameter type="System.String" name="openApiPath" #>
<#@ parameter type="System.String" name="apiRoot" #>
<#@ parameter type="System.String" name="apiNamespace" #>
<#@ parameter type="System.String" name="libraryInterfaceNamespace" #>

<#
    //Check file exists
    OpenApiDiagnostic openApiDiagnostic;
    OpenApiDocument openApiDocument;
	
    using (FileStream fileStream = new FileStream(openApiPath, FileMode.Open, FileAccess.Read)) 
    {
        OpenApiStreamReader reader = new OpenApiStreamReader();
		openApiDocument = reader.Read(fileStream, out openApiDiagnostic);
    }
#>
```

Thanks! 🙂

**Comment by mhutch** [comment]
Yes this is a regression. I added support for loading templates into an `AssemblyLoadContext` on .NET Core so the generated code can be unloaded, similar to how it uses an `AppDomain` on .NET Framework.

It looks like you're loading a third party dll which depends on a particular version of `System.Collections.dll` and the assembly loading logic for the `AssemblyLoadContext` isn't handling this correctly.

I'll probably make the `AssemblyLoadContext` opt-in for now?

**Comment by Bosch-Eli-Black** [comment]
Up to you 🙂

**Comment by mhutch** [comment]
This should have been fixed by fc3b7f4185f468376c3aeec418ace0eec776f45b - please reopen if still broken in 2.3.0.
